FakePrototypeStuff/JoystickInputToEvents.cs

```diff
@@ -21,6 +23,23 @@ public void Update()
 	public const int buttonCount = 20;
 	public const int joystickCount = 10;
 	private float[,] m_LastValues = new float[joystickCount, axisCount + buttonCount];
+
+	static string[] s_Axes;
+
+	static JoystickInputToEvents()
+	{
+		int first = 1;
+		int last = 10;
+		s_Axes = new string[(last - first) * joystickCount + 1];
+		for (int device = 0; device < joystickCount; device++)
+		{
+			for (int i = 0; i <= last - first; i++)
+			{
+				var index = device * (last - first) + i;
```

Comment:
Pre-compute (last - first) outside for for loop. Then cache "device * (the pre-computed last-first)" before entering the inner for-loop.

Reply:
Not a critical method, but fine :P